target_postgres/connector.py

```diff
@@ -53,22 +89,48 @@ def get_sqlalchemy_url(self, config: dict) -> str:
             )
             return cast(str, sqlalchemy_url)
 
-    def truncate_table(self, name):
-        """Clear table data."""
-        self.connection.execute(f"TRUNCATE TABLE {name}")
-
-    def drop_table(self, name):
+    def drop_table(self, table: sqlalchemy.Table):
         """Drop table data."""
-        self.connection.execute(f"DROP TABLE {name}")
+        table.drop(bind=self.connection)
+
+    def clone_table(
+        self, new_table_name, table, metadata, connection, temp_table
+    ) -> sqlalchemy.Table:
+        """Clone a table."""
+        new_columns = []
+        for column in table.columns:
+            new_columns.append(
+                sqlalchemy.Column(
+                    column.name,
+                    column.type,
+                )
+            )
+        if temp_table is True:
+            new_table = sqlalchemy.Table(
+                new_table_name, metadata, *new_columns, prefixes=["TEMPORARY"]
+            )
+        else:
+            new_table = sqlalchemy.Table(new_table_name, metadata, *new_columns)
+        new_table.create(bind=connection)
+        select = sqlalchemy.sql.select(table.columns)
+        insert = Insert(new_table).from_select(names=table.columns, select=select)
```

Comment:
Major bug here, we only want to clone the metadata of the table. Might be easier to just not worry about the clone step here, and just create a temp table